subsys/net/lib/lwm2m/lwm2m_obj_event_log.c

```diff
@@ -0,0 +1,125 @@
+/**
+ * @file lwm2m_event_log.c
+ * @brief
+ *
+ * Copyright (c) 2022 Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Event Log
+ * https://raw.githubusercontent.com/OpenMobileAlliance/lwm2m-registry/prod/20.xml
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_event_log
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <zephyr/init.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_obj_event_log.h"
+
+#define EVENT_LOG_VERSION_MAJOR 1
+#define EVENT_LOG_VERSION_MINOR 0
+#define EVENT_LOG_MAX_ID 6
+
+/*
+ * Calculate resource instances as follows:
+ * start with EVENT_LOG_MAX_ID
+ * subtract EXEC resources (2)
+ */
+#define RESOURCE_INSTANCE_COUNT (EVENT_LOG_MAX_ID - 2)
+
+struct lwm2m_event_log_obj {
+	uint8_t log_class;
+	uint8_t log_status;
+	uint8_t log_data_format;
+};
+
+static struct lwm2m_engine_obj lwm2m_event_log;
+static struct lwm2m_engine_obj_field fields[] = {
+	OBJ_FIELD_DATA(LWM2M_EVENT_LOG_CLASS_ID, R_OPT, U8),
```

Comment:
The object spec says this should be a RW resource.

Reply:
You are totally right: add write attribute